c10/core/TensorImpl.h

```diff
@@ -254,17 +255,21 @@ struct C10_API ExtraMeta {
     if (other.custom_data_ptr_error_msg_) {
       custom_data_ptr_error_msg_ = other.custom_data_ptr_error_msg_;
     }
+    if (other.custom_storage_error_msg_) {
+      custom_storage_error_msg_ = other.custom_storage_error_msg_;
+    }
   }
 
   ExtraMeta(
       std::unique_ptr<c10::SymbolicShapeMeta> symbolic_shape_meta,
       std::unique_ptr<c10::NamedTensorMetaInterface> named_tensor_meta,
       intrusive_ptr<c10::BackendMeta> backend_meta,
-      c10::optional<std::string> custom_data_ptr_error_msg = c10::nullopt)
+      c10::optional<std::string> custom_data_ptr_error_msg = c10::nullopt,
+      c10::optional<std::string> custom_storage_access_error_msg = c10::nullopt)
```

Comment:
why not const char* instead of std::string?

Reply:
this is the owner of the string, char * address might be temporary.. but i could be wrong here ?